File: oj-evaluator/test/memory/mle.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    std::vector<std::vector<int>> bigData;

    const size_t outerSize = 4000000;
    const size_t innerSize = 100;

    bigData.reserve(outerSize);

    for (size_t i = 0; i < outerSize; ++i) {
        bigData.emplace_back(innerSize, static_cast<int>(i));
    }

    cout << "end";

    return 0;
}

File: oj-evaluator/test/memory/mle.java
import java.util.ArrayList;
import java.util.Arrays;

public class mle {
    public static void main(String[] args) {
        System.out.println("Starting massive memory allocation...");

        ArrayList<int[]> bigData = new ArrayList<>();

        final int outerSize = 40_000_00;
        final int innerSize = 100;

        for (int i = 0; i < outerSize; i++) {
            int[] arr = new int[innerSize];
            Arrays.fill(arr, i);
            bigData.add(arr);
        }

        System.out.println("Allocation complete. Press Enter to exit...");
    }
}